motoman_driver/MotoPlus/Controller.c

```diff
@@ -538,7 +543,13 @@ BOOL Ros_Controller_IsMotionReady(Controller* controller)
 BOOL Ros_Controller_IsPflActive(Controller* controller)
 {
 #if (YRC1000||YRC1000u)
-	return ((controller->ioStatus[IO_ROBOTSTATUS_PFL_STOP] | controller->ioStatus[IO_ROBOTSTATUS_PFL_ESCAPE]) != 0);
+	//return ((controller->ioStatus[IO_ROBOTSTATUS_PFL_STOP] | controller->ioStatus[IO_ROBOTSTATUS_PFL_ESCAPE]) != 0);
+	if (controller->bPFLduringRosMove | controller->ioStatus[IO_ROBOTSTATUS_PFL_STOP] | controller->ioStatus[IO_ROBOTSTATUS_PFL_ESCAPE] |
+		( controller->ioStatus[IO_ROBOTSTATUS_PFL_AVOIDING]
+		&& (controller->ioStatus[IO_ROBOTSTATUS_PFL_AVOID_JOINT] | controller->ioStatus[IO_ROBOTSTATUS_PFL_AVOID_TRANS])) != 0)

```

Comment:
is this using binary OR on purpose (ie: `|`) or should it be `||` instead? It seems to mix booleans and bit manipulation.

In the end it probably all gets processed correctly by the compiler, but it would be good to be careful to use the correct operators.

Reply:
I'll make the correction

Reply:
I've pushed the correction in 1fd459a and b7ed970